drivers/fuel_gauge/max17048/max17048.c

```diff
@@ -62,39 +67,57 @@ int max17048_adc(const struct device *i2c_dev, uint16_t *response) {
 }
 
 /* Battery voltage */
-int max17048_voltage(const struct device *i2c_dev, float *response) {
-	uint16_t data;
-	int rc = max17048_adc(i2c_dev, &data);
+int max17048_voltage(const struct device *i2c_dev, uint16_t *response) {
+	int rc = max17048_adc(i2c_dev, response);
 	if (rc < 0){
 		return rc;
 	}
-	*response = (float)data * 78.125 / 1000000;
+	/* Once the value is read, it has to be converted to volts. The datasheet
+	https://www.analog.com/media/en/technical-documentation/data-sheets/MAX17048-MAX17049.pdf
+	Page 10, Table 2. Register Summary: 78.125µV/cell
+	Max17048 only supports one cell so we just have to multiply the value by 78.125 to obtain µV and
+	then divide the value to obtain V.
+
+	But to avoid floats, instead of using 78.125 we will use 78125 and use this value as milli volts
+	instead of volts
+	*/
+
+	*response = *response * 78125 / 1000000;
 	return 0;
 	
 }
 
 /* Battery percentage */
-int max17048_percent(const struct device *i2c_dev, float *response) {
+int max17048_percent(const struct device *i2c_dev, uint8_t *response) {
 	uint16_t data;
 	int rc = max17048_read_register(i2c_dev, REGISTER_SOC, &data);
 	if (rc < 0) {
 		return rc;
 	}
-	*response = (float) data;
-
-	*response /= 256;
+	/* Once the value is read, it has to be converted to percentage. The datasheet
+	https://www.analog.com/media/en/technical-documentation/data-sheets/MAX17048-MAX17049.pdf
+	Page 10, Table 2. Register Summary: 1%/256
+	So to obtain the total percentaje we just divide the read value by 256
+	*/
+	*response = data / 256;
 	return 0;
 }
 
 /* % per hour, positive is charging or negative if discharging */
-int max17048_crate(const struct device *i2c_dev, float *response) {
-	int16_t value;
-	int rc = max17048_read_register(i2c_dev, REGISTER_CRATE, &value);
+int max17048_crate(const struct device *i2c_dev, int16_t *response) {
+	int rc = max17048_read_register(i2c_dev, REGISTER_CRATE, response);
 	if (rc < 0) {
 		return rc;
 	}
 
-	*response = (float)value * 0.208f;
+	/* Once the value is read, it has to ve converted to something useful. The datasheet
+	https://www.analog.com/media/en/technical-documentation/data-sheets/MAX17048-MAX17049.pdf
+	Page 11, Table 2. Register Summary (continued): 0.208%/hr
+	
+	To avoid floats, the value will be multiplied by 208 instead of 0.208, taking into account
+	that the value will ve 1000 times higher
```

Comment:
nit: `ve` to `be` :stuck_out_tongue:

Reply:
Oh, I checked it twice. Looks that I'm blind